app/helpers/window.js

```diff
@@ -15,17 +15,43 @@ const autoResize = (dynamicWindow) => {
     }
   }
 
-  dynamicWindow.webContents.on('did-finish-load', () => {
-    const updateHeight = () => {
-      if (!dynamicWindow) { return }
-      dynamicWindow.webContents.executeJavaScript('document.body.children[0].offsetHeight', (mainContentHeight) => {
-        resize(mainContentHeight)
-      })
+  const updateHeight = () => {
+    if (!dynamicWindow) { return }
+    dynamicWindow.webContents.executeJavaScript('document.body.children[0].offsetHeight', (mainContentHeight) => {
+      resize(mainContentHeight)
+    })
+  }
+
+  let updateHeightIntervalId = null
+  const clearUpdateHeightInterval = () => {
+    if (updateHeightIntervalId) {
+      clearInterval(updateHeightIntervalId)
+      updateHeightIntervalId = null
+    }
+  }
+
+  // register updateHeight interval only when dynamicWindow is visible
+  const registerUpdateHeightInterval = () => {
+    clearUpdateHeightInterval()
+    if (dynamicWindow.isVisible()) {
+      // avoid 125ms delay and redraw the window right away
+      updateHeight()
+      updateHeightIntervalId = setInterval(updateHeight, 125)
     }
-    const id = setInterval(updateHeight, 125)
-    dynamicWindow.on('closed', () => {
-      clearInterval(id)
+  }
+
+  dynamicWindow.webContents.on('did-finish-load', () => {
+    // remove the interval as soon as the dynamicWindow is not visible or destroyed
+    dynamicWindow.on('closed', clearUpdateHeightInterval)
```

Comment:
Is there a reason for the lines 45-51 to be inside of the `did-finish-load` method?